test/realm_test.dart

```diff
@@ -1150,4 +1150,163 @@ Future<void> main([List<String>? args]) async {
       expect(read, [person]);
     });
   });
+  group('IsValid tests:', () {
+    test('RealmObject isValid only if realm is opened', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      var team = Team("team one");
+      expect(team.isValid, true);
+      realm.write(() {
+        realm.add(team);
+      });
+      expect(team.isValid, true);
+      realm.close();
+      expect(team.isValid, false);
+    });
+
+    test('List isValid only if realm is opened', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      realm.write(() {
+        realm.add(Team("Speed Team", players: [
+          Person("Michael Schumacher"),
+          Person("Sebastian Vettel"),
+          Person("Kimi Räikkönen"),
+        ]));
+      });
+
+      var teams = realm.all<Team>();
+
+      expect(teams, isNotNull);
+      expect(teams.length, 1);
+      RealmList<Person> teampPlayers = teams[0].players as RealmList<Person>;
+      expect(teampPlayers.isValid, true);
+      realm.close();
+      expect(teampPlayers.isValid, false);
+    });
+
+    test('Access results after realm closed', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      var team = Team("TeamOne");
+      realm.write(() => realm.add(team));
+      var teams = realm.all<Team>();
+      realm.close();
+      expect(() {
+        teams[0];
+      }, throws<RealmException>("Access to invalidated Results objects"));
+    });
+
+    test('Access deleted object', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      var team = Team("TeamOne");
+      realm.write(() => realm.add(team));
+      var teams = realm.all<Team>();
+      var teamBeforeDelete = teams[0];
+      realm.write(() => realm.delete(team));
+      expect(team.isValid, false);
+      expect(teamBeforeDelete.isValid, false);
+      expect(team, teamBeforeDelete);
+      expect(() {
+        team.name;
+      }, throws<RealmException>("Accessing object of type Team which has been invalidated or deleted"));
+      expect(() {
+        teamBeforeDelete.name;
+      }, throws<RealmException>("Accessing object of type Team which has been invalidated or deleted"));
+    });
+
+    test('Access deleted object collection', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      var team = Team("TeamOne");
+      realm.write(() => realm.add(team));
+      var teams = realm.all<Team>();
+      realm.write(() => realm.delete(team));
+      expect(() {
+        team.players;
```

Comment:
```suggestion
        () => team.players
```

Reply:
Done in all the places in tests methods - commit https://github.com/realm/realm-dart/pull/183/commits/37b38f04f2f52ad09790406dfcbc2aa6a26f0926